scvi/dataset/dataset.py

```diff
@@ -1528,6 +1534,20 @@ def concatenate_arrays(arrays):
     return concatenation
 
 
+def check_nonnegative_integers(X: Union[np.ndarray, sp_sparse.csr_matrix]):
+    """Checks values of X to ensure it is count data"""
+
+    data = X if type(X) is np.ndarray else X.data
```

Comment:
how does this deal with sparsity ? is this why there is the X.data? If so, does it make it dense ? That would be a problem

Reply:
ah ok got it, sounds good to me